### PR TITLE
Apply renovate patches and update the project to support netbox-docker 3.0.1

### LINKED ISSUE #16: Support netbox-docker 3.0
[netbox-docker 3.0 was released last week](https://github.com/netbox-community/netbox-docker/releases/tag/3.0.0). [And another patch release shortly after.](https://github.com/netbox-community/netbox-docker/releases/tag/3.0.1)

This project needs an overhaul to support the changes made to the underlying compose file and containers.

I'm about to start working on this in a fork, I can submit a PR when done.

### PR DESCRIPTION
**Pull request by jsenecal**
#### Breaking changes
- Removal of the now unsupported `netbox_napalm_*` variables
- Uses valkey instead of redis due to licence changes

#### Features
- Replaces `netbox_proxy_network_name` (a string) with `netbox_additionnal_network_names` (a list) to allow for different scenarios
- Updates the docker-compose file to add upstream health checks.

Fixes #16 and #6

### CODE REVIEW

**Comment by jsenecal** [comment]
Fixes #16

File: README.md [hunk 1]
```diff
@@ -65,21 +65,18 @@ Configuration and installation options are made available as variables. Some of
...
-| `netbox_redis_cache_ssl`                      | `false`                       | If true, communication with redis is secured with SSL                                                                                                   |
-| `netbox_redis_host`                           | `redis`                       | Redis instance host. This should not be changed if using the default docker-compose setup.                                                              |
-| `netbox_redis_insecure_skip_tls_verify`       | `false`                       | If true, certificates for redis are not checked                                                                                                         |
-| `netbox_redis_password`                       | `H733Kdjndks81`               | Redis instance password                                                                                                                                 |
+| `netbox_additionnal_network_names`                   | -                             | Extra external networks to attach to the netbox container. See [Traefik/nginx-proxy Support](#traefiknginx-proxy-support)                                 |
```
Comment (wastrachan) [inline README.md]:
```suggestion
| `netbox_additional_network_names`                   | -                             | Extra external networks to attach to the netbox container. See [Traefik/nginx-proxy Support](#traefiknginx-proxy-support)                                 |
```

File: README.md [hunk 2]
```diff
@@ -96,13 +93,14 @@ Including an example of how to use your role (for instance, with variables passe
 
 ## Traefik/nginx-proxy Support
 
-This playbook can be used with [traefik](https://hub.docker.com/_/traefik) or jwilder's [nginx-proxy](https://hub.docker.com/r/jwilder/nginx-proxy) by adding labels with `netbox_container_labels`, or environment variables with `netbox_container_env`, respectively. Additionally, `netbox_proxy_network_name` will attach the netbox service to an additional network, as traefik/nginx-proxy usually reside in a network other than that created by docker-compose projects. While not a complete guide to these services, your configuration may look like the below:
+This playbook can be used with [traefik](https://hub.docker.com/_/traefik) or jwilder's [nginx-proxy](https://hub.docker.com/r/jwilder/nginx-proxy) by adding labels with `netbox_container_labels`, or environment variables with `netbox_container_env`, respectively. Additionally, `netbox_additionnal_network_names` will attach the netbox service to additional networks, as traefik/nginx-proxy usually reside in a network other than that created by docker-compose projects. While not a complete guide to these services, your configuration may look like the below:
```
Comment (wastrachan) [inline README.md]:
```suggestion
This playbook can be used with [traefik](https://hub.docker.com/_/traefik) or jwilder's [nginx-proxy](https://hub.docker.com/r/jwilder/nginx-proxy) by adding labels with `netbox_container_labels`, or environment variables with `netbox_container_env`, respectively. Additionally, `netbox_additional_network_names` will attach the netbox service to additional networks, as traefik/nginx-proxy usually reside in a network other than that created by docker-compose projects. While not a complete guide to these services, your configuration may look like the below:
```

File: README.md [hunk 3]
```diff
@@ -96,13 +93,14 @@ Including an example of how to use your role (for instance, with variables passe
...
 
 #### traefik
 
 ```yaml
 netbox_port: null
-netbox_proxy_network_name: "default"
+netbox_additionnal_network_names: 
+  - traefik
```
Comment (wastrachan) [inline README.md]:
```suggestion
netbox_additional_network_names: 
  - traefik
```

File: README.md [hunk 4]
```diff
@@ -113,7 +111,8 @@ netbox_container_labels:
 
 ```yaml
 netbox_port: null
-netbox_proxy_network_name: "default"
+netbox_additionnal_network_names:
```
Comment (wastrachan) [inline README.md]:
```suggestion
netbox_additional_network_names: 
```

File: templates/docker-compose.yml.j2 [hunk 5]
```diff
@@ -34,36 +38,42 @@ services:
     {% endif %}
     networks:
       - default
-    {% if netbox_proxy_network_name %}
-      - proxy
+    {% if netbox_additionnal_network_names %}
```
Comment (wastrachan) [inline templates/docker-compose.yml.j2]:
```suggestion
    {% if netbox_additional_network_names %}
```

File: templates/docker-compose.yml.j2 [hunk 6]
```diff
@@ -34,36 +38,42 @@ services:
...
+      {% for network in netbox_additionnal_network_names %}- {{ network }}{% endfor %}
```
Comment (wastrachan) [inline templates/docker-compose.yml.j2]:
```suggestion
      {% for network in netbox_additional_network_names %}- {{ network }}{% endfor %}
```

File: templates/docker-compose.yml.j2 [hunk 7]
```diff
@@ -73,42 +83,57 @@ services:
...
   netbox-media-files:
     driver: local
   netbox-postgres-data:
     driver: local
-  netbox-redis-data:
+  netbox-valkey-data:
     driver: local
 
-{% if netbox_proxy_network_name %}
+{% if netbox_additionnal_network_names %}
```
Comment (wastrachan) [inline templates/docker-compose.yml.j2]:
```suggestion
{% if netbox_additional_network_names %}
```

File: templates/docker-compose.yml.j2 [hunk 8]
```diff
@@ -73,42 +83,57 @@ services:
...
 networks:
-  proxy:
-    name: {{ netbox_proxy_network_name }}
+{% for network in netbox_additionnal_network_names %}
```
Comment (wastrachan) [inline templates/docker-compose.yml.j2]:
```suggestion
{% for network in netbox_additional_network_names %}
```

**Review by wastrachan — CHANGES_REQUESTED** [review]
Thanks for contributing all of this work, I appreciate all of the updates and the generic re-implementation of external networks. 

I'd like to understand the switch from Redis a little better. I'm aware of their license changes, as well as the impact they have on the SAAS ecosystem, as well as for various MSP uses. I also fully support the right of an individual to choose an alternative open source project when they disagree with a decision and when it suits their needs. 

Is there something about the new Redis license that makes it _incompatible_ with this project, or this project's license? Many of us rely on the stability and proven history of Redis in production workloads, and aren't yet ready to move to another offering that has not yet had ample time to prove it's development practices and stewardship. If there is an incompatibility, I'm of course happy to honor licenses and commit to removing Redis from this project, but if it's just preference- I'd prefer keeping things as they are, or perhaps exploring an option that would let the end user opt for Redis or Valkey as it fits their goals.

**Comment by jsenecal** [comment]
> Is there something about the new Redis license that makes it _incompatible_ with this project, or this project's license? Many of us rely on the stability and proven history of Redis in production workloads, and aren't yet ready to move to another offering that has not yet had ample time to prove it's development practices and stewardship. If there is an incompatibility, I'm of course happy to honor licenses and commit to removing Redis from this project, but if it's just preference- I'd prefer keeping things as they are, or perhaps exploring an option that would let the end user opt for Redis or Valkey as it fits their goals.

Under the new license, cloud service providers hosting Redis offerings will no longer be permitted to use the source code of Redis free of charge. For example, cloud service providers will be able to deliver Redis 7.4 only after agreeing to licensing terms with Redis, the maintainers of the Redis code. These agreements will underpin support for existing integrated solutions and provide full access to forthcoming Redis innovations.

This also means that a service provider cannot freely distribute a solution with a Redis component under this new licensing. 

Valkey is a FOSS fork of Redis backed by the Linux Foundation. 

This change also follows upstream decision to move away from Redis. https://github.com/netbox-community/netbox-docker/pull/1203

**Comment by wastrachan** [comment]
> > Is there something about the new Redis license that makes it _incompatible_ with this project, or this project's license? Many of us rely on the stability and proven history of Redis in production workloads, and aren't yet ready to move to another offering that has not yet had ample time to prove it's development practices and stewardship. If there is an incompatibility, I'm of course happy to honor licenses and commit to removing Redis from this project, but if it's just preference- I'd prefer keeping things as they are, or perhaps exploring an option that would let the end user opt for Redis or Valkey as it fits their goals.
> 
> Under the new license, cloud service providers hosting Redis offerings will no longer be permitted to use the source code of Redis free of charge. For example, cloud service providers will be able to deliver Redis 7.4 only after agreeing to licensing terms with Redis, the maintainers of the Redis code. These agreements will underpin support for existing integrated solutions and provide full access to forthcoming Redis innovations.
> 
> This also means that a service provider cannot freely distribute a solution with a Redis component under this new licensing.
> 
> Valkey is a FOSS fork of Redis backed by the Linux Foundation.
> 
> This change also follows upstream decision to move away from Redis. [netbox-community/netbox-docker#1203](https://github.com/netbox-community/netbox-docker/pull/1203)

Thanks, this is helpful. I generally undestand the ramifications of the new license-- but knowing that Netbox has switched offers me at least the assurance that Nextbox will be continuously tested with Valkey.

**Review by wastrachan — APPROVED** [review]

**Pull request merged**